src/views/sources/viewSourcesList.tsx

```diff
@@ -421,25 +400,33 @@ const SourcesListView: React.FunctionComponent = () => {
         </ConditionalTableBody>
       </Table>
       <Pagination variant="bottom" widgetId="server-paginated-example-pagination" />
-      {!!credentialsSelected.length && (
-        <Modal
-          variant={ModalVariant.small}
-          title={t('view.label', { context: 'credentials' })}
-          isOpen={!!credentialsSelected}
-          onClose={() => setCredentialsSelected([])}
-          actions={[
-            <Button key="cancel" variant="secondary" onClick={() => setCredentialsSelected([])}>
-              Close
-            </Button>
-          ]}
-        >
-          <List isPlain isBordered>
-            {credentialsSelected.map(c => (
-              <ListItem key={c.name}>{c.name}</ListItem>
-            ))}
-          </List>
-        </Modal>
-      )}
+      <Modal
+        variant={ModalVariant.small}
+        title={t('view.label', { context: 'credentials' })}
+        isOpen={credentialsSelected.length > 0}
+        onClose={() => setCredentialsSelected([])}
+        actions={[
+          <Button
+            key="confirm"
+            variant="danger"
+            onClick={() => {
+              // Add your specific action if needed
+              setCredentialsSelected([]);
+            }}
+          >
+            {t('table.label', { context: 'close' })}
+          </Button>,
+          <Button key="cancel" variant="link" onClick={() => setCredentialsSelected([])}>
+            {t('form-dialog.label', { context: 'cancel' })}
+          </Button>
+        ]}
+      >
+        <List isPlain isBordered>
+          {credentialsSelected.map(c => (
+            <ListItem key={c.name}>{c.name}</ListItem>
+          ))}
+        </List>
+      </Modal>
```

Comment:
future: 
- this modal should go on a list of getting it's own component, like some of the other similar components and modals
- and we need to implement view context when we make new components... hopefully letting us avoid "prop drilling" ... feel free to look that one up =)

Reply:
Added a TODO comment for the future